src/Discord.Net.Commands/Builders/SlashCommandBuilder.cs

```diff
@@ -0,0 +1,491 @@
+                if (!Regex.IsMatch(value, @"^[\w-]{3,32}$"))
+                    throw new ArgumentException("Option name cannot contian any special characters or whitespaces!");
+
+                _name = value;
+            }
+        }
+
+        /// <summary>
+        ///     The description of this option.
+        /// </summary>
+        public string Description
+        {
+            get => _description;
+            set
+            {
+                if (value?.Length > SlashCommandBuilder.MaxDescriptionLength)
+                    throw new ArgumentException("Description length must be less than or equal to 100");
+                if (value?.Length < 1)
+                    throw new ArgumentException("Name length must at least 1 character in length");
+
+                _description = value;
+            }
+        }
+
+        /// <summary>
+        ///     The type of this option.
+        /// </summary>
+        public ApplicationCommandOptionType Type { get; set; }
+
+        /// <summary>
+        ///     The first required option for the user to complete. only one option can be default.
+        /// </summary>
+        public bool Default { get; set; }
+
+        /// <summary>
+        ///     <see langword="true"/> if this option is required for this command, otherwise <see langword="false"/>.
+        /// </summary>
+        public bool Required { get; set; }
+
+        /// <summary>
+        ///     choices for string and int types for the user to pick from.
+        /// </summary>
+        public List<ApplicationCommandOptionChoiceProperties> Choices { get; set; }
+
+        /// <summary>
+        ///     If the option is a subcommand or subcommand group type, this nested options will be the parameters.
+        /// </summary>
+        public List<SlashCommandOptionBuilder> Options { get; set; }
+
+        /// <summary>
+        ///     Builds the current option.
+        /// </summary>
+        /// <returns>The build version of this option</returns>
+        public ApplicationCommandOptionProperties Build()
+        {
+            bool isSubType = this.Type == ApplicationCommandOptionType.SubCommand || this.Type == ApplicationCommandOptionType.SubCommandGroup;
+
+            if (isSubType && (Options == null || !Options.Any()))
+                throw new ArgumentException(nameof(Options), "SubCommands/SubCommandGroups must have at least one option");
+
+            if (!isSubType && (Options != null && Options.Any()))
+                throw new ArgumentException(nameof(Options), $"Cannot have options on {Type} type");
+
+            return new ApplicationCommandOptionProperties()
+            {
+                Name = this.Name,
+                Description = this.Description,
+                Default = this.Default,
+                Required = this.Required,
+                Type = this.Type,
+                Options = new List<ApplicationCommandOptionProperties>(this.Options.Select(x => x.Build())),
```

Comment:
I think you should add a simple null check on this line. In my testing i found out "Linq" throws an error while trying to add a sub-command's parameters, which cannot have any "application command options".

Something like this should do the trick:
```csharp
Options = Options != null ? new List<ApplicationCommandOptionProperties>(this.Options.Select(x => x.Build())) : null,
```

Reply:
Null check has been added [Here](https://github.com/discord-net/Discord.Net/blob/f4a8b219ed6d9de643154a9ea7f6b34e8591f984/src/Discord.Net.Commands/Builders/SlashCommandBuilder.cs#L393)